api/area_test.go

```diff
@@ -198,6 +215,52 @@ func TestGetAreaRelationshipsReturnsOk(t *testing.T) {
 	})
 }
 
+func TestGetAreaRelationshipsWithParameterReturnsOk(t *testing.T) {
+	Convey("Given a successful request for child area relationship data - E92000001", t, func() {
+		r := httptest.NewRequest(http.MethodGet, fmt.Sprintf("http://localhost:2200/v1/areas/%s/relations?relationship=child", YorkshireAreaData), nil)
+		r.Header.Set(models.AcceptLanguageHeaderName, "en")
+		w := httptest.NewRecorder()
+		childRelatedAreas := []*models.AreaBasicData{
+			{SheffieldAreaData, SheffieldName},
+		}
+		relatedAreas := []*models.AreaBasicData{
+			{"E92000001", "North East"},
+			{"E12000002", "North West"},
+			{SheffieldAreaData, SheffieldName},
+		}
+
+		expectedChildRelationShips := []*models.AreaRelationShips{
+			{SheffieldAreaData, SheffieldName, "/v1/area/E08000019"},
+		}
+
+		areaApi, _ := GetAPIWithRDSMocks(&mock.RDSAreaStoreMock{
+			ValidateAreaFunc: func(areaCode string) error {
+				return nil
+			},
+			GetRelationshipsFunc: func(areaCode, relationshipParameter string) ([]*models.AreaBasicData, error) {
+				if relationshipParameter == "child" {
+					return childRelatedAreas, nil
+				}
+				return relatedAreas, nil
+			},
+		})
+		areaApi.Router.ServeHTTP(w, r)
+
+		Convey("When request child area relationship data is served", func() {
```

Comment:
Can you add test(s) to cover the handling of error responses too if they don't exist in the test?